### PR TITLE
Fixes search passthrough to Mirador

### PR DESCRIPTION
**Pull request by mspalti**
## Description
The mirador Angular component should pass DSpace discovery query terms to the Mirador viewer whenever the previous route in history was a search request. I'm not sure why this stopped working. The PR fixes the problem by simplifying the utility function that checks for a search and returns the query terms (if found). The expected behavior is that the Mirador viewer will be initialized with search results present.  New tests are provided.

## Instructions for Reviewers
Testing this is difficult at the moment because it requires installing and populating a separate Solr core and indexing OCR data from Item Bitstreams (in ALTO, hOCR, or MiniOcr). That's possible if you have some data available. Also, some tooling  and a preconfigured core are under review  for a later release in this PR [#8241](https://github.com/DSpace/DSpace/pull/8241). Some folks will be working with the experimental IIIF search support so I would be helpful to get this minor Angular fix added to 7.3.  

List of changes in this PR:
* Small update to `item-iiif-utils` function.
* Adds tests

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [ ] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [ ] My PR doesn't introduce circular dependencies
- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by abollini — CHANGES_REQUESTED** [review]
hi @mspalti 
it looks to work for me in the basic scenarios but I would like to get your feedback on my two inline comments and see if the suggested code/test improvements are feasible

File: src/app/item-page/simple/item-types/shared/item-iiif-utils.ts [hunk 1]
```diff
@@ -26,10 +26,9 @@ export const getDSpaceQuery = (item: Item, routeService: RouteService): Observab
       return r.includes('/search');
     }),
     map(r => {
-      const arr = r.split('&');
-      const q = arr[1];
-      const v = q.split('=');
+      const v = r.split('query=');
```
[the remark this answers is not in the document]
Comment (mwoodiupui) [reply src/app/item-page/simple/item-types/shared/item-iiif-utils.ts:29]: Use `const u = new URL(r)` and find what you want in `u.searchParams`?  https://stackoverflow.com/a/24006120/2916377

Agreed:  one should always think long and hard before writing one of these ad-hoc parsers for something as complex as a URL.

**Comment by mspalti** [comment]
I just tested this in a development environment. The query DSpace query passthrough worked fine. I believe this is ready.

**Review by abollini — APPROVED** [review]
hi @mspalti unfortunately, my re-review and approval was never submitted...
there is a small conflict right now, can you resolve it so that this can be merged?

**Comment by tdonohue** [comment]
@mspalti : If you have any time in the coming days, this can be included in 7.3 if the merge conflicts are resolved before Tuesday. After Tuesday, I'll likely have to reschedule this for 7.4.  Thanks!

**Comment by mspalti** [comment]
@tdonohue , @abollini , this should be ready to merge.

**Pull request merged**